### PR TITLE
scripts: update build-aci

### PR DESCRIPTION
**Pull request by alban**
Building App Container Image for tests with Rocket was not working for me. The two patches are fixing that with static compilation and changes in the build-aci script.

Please let me know if it breaks your workflow.

### CODE REVIEW

File: scripts/build-aci [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+
+# A non-installed actool can be used, for example:
+# ACTOOL=../../../appc/spec/bin/actool
+ACTOOL=${ACTOOL:-actool}
+
+IMAGEDIR=${IMAGEDIR:-image-aci}
+
+VERSION=$1
+
+if [ ! -x ./etcd ] ; then
```
Comment (jonboulle) [inline scripts/build-aci]: I would rather this was invoked from the repository root

File: scripts/build-aci [hunk 2]
```diff
@@ -1,11 +1,29 @@
-#!/usr/bin/bash
+#!/bin/bash
+
+# A non-installed actool can be used, for example:
+# ACTOOL=../../../appc/spec/bin/actool
+ACTOOL=${ACTOOL:-actool}
```
Comment (jonboulle) [inline scripts/build-aci]: -x on $ACTOOL too?

Comment (alban) [inline scripts/build-aci]: I will add that test, and also check that $IMAGEDIR is an empty writeable directory.

File: scripts/build-aci (hunk 1 above)
Comment (alban) [inline scripts/build-aci]: Agree, I will do that :)

**Comment by alban** [comment]
Branch updated with better checks.

**Comment by kelseyhightower** [comment]
@alban This looks awesome.

File: scripts/build-aci [hunk 3]
```diff
@@ -25,9 +56,9 @@ cat <<DF > manifest
...
 
-mkdir -p rootfs/etc/
-cat <<DF > rootfs/etc/hosts
+mkdir -p $IMAGEDIR/rootfs/etc/
+cat <<DF > $IMAGEDIR/rootfs/etc/hosts
 127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
 DF
 
-actool build -overwrite=true . ../etcd-${1}-linux-amd64.aci
+$ACTOOL build -overwrite=true $IMAGEDIR etcd-${1}-linux-amd64.aci
```
Comment (jonboulle) [inline scripts/build-aci]: Could you put the output file in bin/ please?

File: build [hunk 4]
```diff
@@ -11,7 +11,8 @@ ln -s ${PWD} $GOPATH/src/${REPO_PATH}
 
 eval $(go env)
 
-go build -o bin/etcd ${REPO_PATH}
+# Static compilation is useful when etcd is ran in a container
```
Comment (jonboulle) [inline build]: s/ran/run/

File: scripts/build-aci [hunk 5]
```diff
@@ -1,11 +1,42 @@
...
+    exit 1
 fi
 
-cat <<DF > manifest
+if [ -n "$(ls -A $IMAGEDIR/rootfs)" ]; then
+    echo "$IMAGEDIR/rootfs is not empty"
+    exit 1
+fi
+
+cp bin/* $IMAGEDIR/rootfs
```
Comment (jonboulle) [inline scripts/build-aci]: Rather explicitly list the files. For reference, this is the current etcd aci:

```
drwxr-xr-x 501/20            0 2014-12-18 10:51 rootfs
drwxr-xr-x 501/20            0 2014-12-18 10:32 rootfs/Documentation
-rw-r--r-- 501/20         2266 2014-12-18 10:32 rootfs/Documentation/0_4_migration_tool.md
-rw-r--r-- 501/20         6770 2014-12-18 10:32 rootfs/Documentation/admin_guide.md
-rw-r--r-- 501/20        25262 2014-12-18 10:32 rootfs/Documentation/api.md
-rw-r--r-- 501/20         3105 2014-12-18 10:32 rootfs/Documentation/backward_compatibility.md
-rw-r--r-- 501/20        11683 2014-12-18 10:32 rootfs/Documentation/clustering.md
-rw-r--r-- 501/20         4598 2014-12-18 10:32 rootfs/Documentation/configuration.md
-rw-r--r-- 501/20          753 2014-12-18 10:32 rootfs/Documentation/glossary.md
-rw-r--r-- 501/20         3303 2014-12-18 10:32 rootfs/Documentation/other_apis.md
-rw-r--r-- 501/20         2751 2014-12-18 10:32 rootfs/Documentation/proxy.md
-rw-r--r-- 501/20         7053 2014-12-18 10:32 rootfs/Documentation/runtime-configuration.md
-rw-r--r-- 501/20         5186 2014-12-18 10:32 rootfs/README-etcdctl.md
-rw-r--r-- 501/20         4788 2014-12-18 10:32 rootfs/README.md
drwxr-xr-x 501/20            0 2014-12-18 10:51 rootfs/etc
-rw-r--r-- 501/20           79 2014-12-18 10:51 rootfs/etc/hosts
-rwxr-xr-x 501/20      9159736 2014-12-18 10:32 rootfs/etcd
-rwxr-xr-x 501/20      7199704 2014-12-18 10:32 rootfs/etcd-migrate
-rwxr-xr-x 501/20      6266752 2014-12-18 10:32 rootfs/etcdctl
-rw-r--r-- root/root       374 2014-12-18 10:51 manifest
```

I think the documentation/readme could be useful to keep in there for now. We definitely don't need `etcd-migrate`, though.

Comment (jonboulle) [inline scripts/build-aci]: (recursive copy of documentation is fine)

**Comment by jonboulle** [comment]
Thanks - just a few other small things I didn't catch in my first pass, sorry.

**Comment by alban** [comment]
@jonboulle : thanks for the review. I pushed a new version of the branch.

File: scripts/build-aci [hunk 6]
```diff
@@ -1,11 +1,45 @@
...
+    echo "bin/etcd not found. Is it compiled?"
+    exit 1
+fi
+
+if [ -z "$VERSION" ] ; then
+    echo "Usage: scripts/build-aci VERSION"
+    exit 1
+fi
+
+mkdir -p $IMAGEDIR/rootfs
```
Comment (jonboulle) [inline scripts/build-aci]: Honestly we should just remove this directly. This is OK for now though.

**Comment by jonboulle** [comment]
@alban thanks for the improvements!
